src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -60,22 +62,27 @@ namespace AppInstaller::Repository::Rest::Schema::V1_0
 
         utility::string_t GetSearchEndpoint(const std::string& restApiUri)
         {
-            std::string fullSearchAPI = GetRestAPIBaseUri(restApiUri) + ManifestSearchPostEndpoint;
+            std::string fullSearchAPI = GetRestAPIBaseUri(restApiUri) + ManifestSearchPostEndpoint.data();
             utility::string_t searchAPI = utility::conversions::to_string_t(fullSearchAPI);
             return searchAPI;
         }
 
-        utility::string_t GetManifestByVersionEndpoint(const std::string& restApiUri, const std::string& packageId, const std::string& version)
+        utility::string_t GetManifestByVersionEndpoint(
+            const std::string& restApiUri, const std::string& packageId, const std::string& version, const std::string& channel)
         {
-            // TODO: Replace with manifest version endpoint 
-            std::string versionEndpoint = GetRestAPIBaseUri(restApiUri) + "/api/packages/" + packageId + "/versions/" + version;
+            std::string versionEndpoint = GetRestAPIBaseUri(restApiUri) + ManifestByVersionAndChannelGetEndpoint.data() + packageId + "?version=" + version + "&channel=" + channel;

```

Comment:
I think you followed my model too closely with the SQLite interface.  I separated the lifetime of the connection from the interface used to access it for arguable reasons.  But you have no such concern here.  I think the only concern you have is that `Create` on the factory might want to determine the interface version to use (and at the same time will ensure that the endpoint exists), but there is no (complex) lifetime to need to manage.

I would suggest making the interface less static, taking in some data on construction and storing what it needs to operate with rather than requiring every call into to pass in the data repeatedly.

Reply:
Gotcha, that makes sense. Modified to have Rest API uri to avoid sending it repeatedly. 

---
In reply to: [586811608](https://github.com/microsoft/winget-cli/pull/765#discussion_r586811608) [](ancestors = 586811608)